content/authors/client-side-validation-with-react-alert/index.md

```diff
@@ -0,0 +1,366 @@
+
+```js
+const validate = validateSignUp(email, password, pwdConfirm, alert ||" ");
+
+    if (validate) {
+      alert.success("You've signed up successfully. Proceed to login");
+    }
+};
+```
+
+- **Setting up the alert template**
+
+This step involves providing the `react-alert` template API to our application. Without this, the error messages wouldn’t be displayed on the webpage. Let’s start by taking a look at the setup below.
+
+We’d start by creating a component that’d serve as a container for the error messages. In the `utils` folder, we’d add all the logic of the alert-template.
+
+```js
+import React from "react";
+
+import { transitions, positions, Provider as AlertProvider } from "react-alert";
+
+const AlertTemplate = ({ message, options }) => {
+  return (
+    <div className={options.type === "success" ? "msg-success" : "msg-error"}>
+      <p>{message}</p>
+    </div>
+  );
+};
+
+// this component serves as the container
+// that holds/displays the error messages due to the validation
+// script that runs on all the app components that have an
+// input field
+const Message = ({ children }) => {
+  const options = {
+    position: positions.TOP_RIGHT,
+    timeout: 3500,
+    offset: "0px",
+    transition: transitions.SCALE,
+  };
+
+  return (
+    <AlertProvider template={AlertTemplate} {...options}>
+      {children}
+    </AlertProvider>
+  );
+};
+
+export default Message;
+```
+
+This snippet below shows a conditional statement but in a ternary syntax. What this translates to, is: when the `options.type` is “success”, add a CSS class of “msg-success”, if not, add a CSS class of “msg-error”.
+
+The “msg-success” gives the alert template/modal a green background while “msg-error” adds a red background to the template.
+
+Once that is complete, we’d go-ahead to add the template as a parent element in the app component. That can be done by editing the content of `_app.js`
+
+```js
+import React from “react”
+import Message from "../src/utils/Message";
+import Head from “next/head”
+
+function App({ Component, pageProps }) {
+    return (
+      <React.Fragment>
+       <Head>
+         <link rel="icon" href="#" />
+         <meta
+           name="viewport"
+           content="minimum-scale=1, initial-scale=1, width=device-width"
+         />
+       </Head>
+       <Message>
+          <Component {...pageProps} />
+       </Message>
+      </React.Fragment>
+    );
+}
+
+export default App;
+```
+
+Wrapping it up
+You’d have noticed that we didn’t dive into the styling of this project, that is because I do not want the article to be too long. You can go on ahead to any styling that befits this project. You can still check out react-alert’s [documentation](https://github.com/schiehll/react-alert) so you have a full understanding of how the package works.
```

Comment:
```suggestion
You have noticed we didn’t dive into the styling of this project, that is because I do not want the article to be too long. You can go ahead and style your project to your own liking.

You can also check out react-alert’s [documentation](https://github.com/schiehll/react-alert) so you get a good understanding of how the package works.
```